src/main/java/com/assignment/application/interceptor/AuthInterceptor.java

```diff
@@ -1,51 +1,103 @@
+import com.assignment.application.repo.CompanyRepo;
 import com.assignment.application.repo.EmployeeRepo;
+import com.assignment.application.service.RedisService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Configuration;
 import org.springframework.stereotype.Component;
 import org.springframework.web.servlet.ModelAndView;
 import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Base64;
 
 @Component
 public class AuthInterceptor extends HandlerInterceptorAdapter {
 
     @Autowired
     private EmployeeRepo employeeRepo;
 
+    @Autowired
+    private RedisService redisService;
+
+    @Autowired
+    private CompanyRepo companyRepo;
+
+    private final String START_TIME = "Start time";
+
+    private final Logger LOG = LogManager.getLogger(AuthInterceptor.class);
+
+    private Long companyId = 0L;
+
     @Override
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
-        super.preHandle(request, response, handler);
-        String userName = request.getHeader("username");
-        String password = request.getHeader("password");
-        if(userName==null || password==null || userName.isEmpty() || password.isEmpty()){
-            throw new IllegalArgumentException("Invalid Credential");
+        Long startTime = System.currentTimeMillis();
+        request.setAttribute(START_TIME, startTime);
+        System.out.println(request.getRequestURL());
+        if (request.getMethod().equalsIgnoreCase("POST") &&
+                validateUrl(request.getRequestURL().toString())) {
+            String userName = request.getHeader("username");
+            String password = request.getHeader("password");
+            String decodedPassword = new String(Base64.getDecoder().decode(password));
+            if(userName.equalsIgnoreCase("superadmin") && decodedPassword.equalsIgnoreCase("superadmin")){
+                String checkIfExist = redisService.getKeyValue(StringConstant.ACCESS_TOKEN_GENERATED + userName);
+                if(checkIfExist==null){
+                    return true;
+                }
+                throw new IllegalArgumentException("Invalid Login Credential");
+            }
+            Employee employee = employeeRepo.getEmployee(userName);
+            if(employee==null){
+                throw new NotExistsException("No such employee exists");
+            }
+            String checkIfExist = redisService.getKeyValue(StringConstant.ACCESS_TOKEN_GENERATED + employee.getEmployeeId());
+            if (checkIfExist==null && employee.getEmployeeId().equalsIgnoreCase(userName) &&
+                    employee.getDob().equalsIgnoreCase(decodedPassword) &&
```

Comment:
For now we don't have a db. And I want my password to be encrypted that's why I just decoded it in the code so that it shows the use of encryption. We don't have to decrypt it if we are using a db for storing the password and username but the thing is storing password and username is not what we have do.